.gitignore

```diff
@@ -1,3 +1,6 @@
+# Rendered templates
+deploy/
```

Comment:
You ignore it and added .gitkeep?  Keeping the contents in deploy would let us point to that for an oc apply if someone else wanted to use it.  Else it's use oc process on the template or build it yourself.  All doable.  Not a blocker for this PR, just something I wondered about.

Reply:
Yeah... we need the directory because that's where non-Hive artifacts land, but we do not want them in git. This is a pattern used in other places (which doesn't mean it's good or bad, but it's "the pattern").